crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs

```diff
@@ -625,3 +706,113 @@ async fn test_subnet_id_call(
     context.shutdown().await?;
     Ok(())
 }
+
+/// Test perform send token and check for transaction
+/// in the certificate (by observing target subnets)
+#[rstest]
+#[test(tokio::test)]
+#[serial]
+async fn test_subnet_send_token_processing(
+    #[with(8546)]
+    #[future]
+    context_running_subnet_node: Context,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let context = context_running_subnet_node.await;
+    let test_private_key = hex::decode(TEST_SECRET_ETHEREUM_KEY).unwrap();
+    let subnet_jsonrpc_endpoint = "http://".to_string() + &context.jsonrpc();
+    let subnet_smart_contract_address =
+        "0x".to_string() + &hex::encode(context.i_topos_core.address());
+
+    // Create runtime proxy worker
+    info!("Creating subnet runtime proxy");
+    let mut runtime_proxy_worker = SubnetRuntimeProxyWorker::new(
+        SubnetRuntimeProxyConfig {
+            subnet_id: SOURCE_SUBNET_ID_1,
+            endpoint: context.jsonrpc(),
+            subnet_contract_address: subnet_smart_contract_address.clone(),
+            verifier: 0,
+            source_head_certificate_id: None,
+        },
+        test_private_key.clone(),
+    )
+    .await?;
+    tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
+    info!("Set source head certificate to 0");
+    if let Err(e) = runtime_proxy_worker
+        .set_source_head_certificate_id(Some(CERTIFICATE_ID_1))
+        .await
+    {
+        panic!("Unable to set source head certificate id: {e}");
+    }
+
+    // Deploy token contract
+    let i_erc20 = deploy_test_token(
+        &hex::encode(&test_private_key),
+        &subnet_jsonrpc_endpoint,
+        context.i_topos_messaging.address(),
+    )
+    .await?;
+
+    // Approve token spending
+    if let Err(e) = i_erc20
+        .approve(context.i_topos_messaging.address(), U256::from(10))
+        .legacy()
+        .gas(400000)
+        .send()
+        .await?
+        .await
+    {
+        panic!("Unable to perform token approval {e}");
+    }
+
+    // Perform send token
+    info!("Sending token");
+    if let Err(e) = context
+        .i_topos_messaging
+        .send_token(
+            TARGET_SUBNET_ID_2.into(),
+            "00000000000000000000000000000000000000AA".parse()?,
+            i_erc20.address(),
+            U256::from(2),
+        )
+        .legacy()
+        .gas(5_000_000)
```

Comment:
Same for the gas variables. If we have these as "parameter variables" at the beginning, it's easier to reason and understand?